source/auth/tests/connection-string.yml

```diff
@@ -291,3 +291,11 @@ tests:
         description: "authSource without username is invalid (default mechanism)"
         uri: "mongodb://localhost/?authSource=foo"
         valid: false
+
+        description: "should throw an exception if not username provided"
```

Comment:
Suggest "if no username provided" or "if no username is provided" or "if username is not provided".